src/tooltip.tsx

```diff
@@ -178,84 +249,146 @@ const Tooltip: React.FC<Props> = ({
+                  tooltipClient.width / 2
+                : window.pageXOffset +
+                  targetBoundingClientRect.current.left +
+                  targetBoundingClientRect.current.width / 2 -
+                  width / 2;
+
+            const leftAdjustment = hasSpace('left')
+                ? bottomAdjustment
+                : !width
+                ? targetBoundingClientRect.current.left - tooltipClient.width - distanceToTarget
+                : targetBoundingClientRect.current.left - width - distanceToTarget;
+
+            const topAdjustment = hasSpace('left')
+                ? targetBoundingClientRect.current.top < tooltipClient.height
+                    ? window.pageYOffset + targetBoundingClientRect.current.bottom + distanceToTarget
+                    : window.pageYOffset + targetBoundingClientRect.current.top - distanceToTarget
+                : changedPosition
+                ? changedPosition
+                : window.pageYOffset +
+                  targetBoundingClientRect.current.top +
+                  targetBoundingClientRect.current.height / 2;
+
+            const containerPos = {
+                right: {
+                    left: hasSpace('right')
+                        ? leftAdjustment
+                        : targetBoundingClientRect.current.right + distanceToTarget,
+                    top: hasSpace('right')
+                        ? topAdjustment
+                        : changedPosition
+                        ? changedPosition
+                        : window.pageYOffset +
+                          targetBoundingClientRect.current.top +
+                          targetBoundingClientRect.current.height / 2,
+                },
+                left: {
+                    left: leftAdjustment,
+                    top: topAdjustment,
+                },
+                top: {
+                    top: hasSpace('right')
+                        ? topAdjustment
+                        : window.pageYOffset + targetBoundingClientRect.current.top - distanceToTarget,
+
+                    left: hasSpace('right') ? leftAdjustment : bottomAdjustment,
+                },
+                bottom: {
+                    top: hasSpace('right')
+                        ? topAdjustment
+                        : window.pageYOffset + targetBoundingClientRect.current.bottom + distanceToTarget,
+                    left: hasSpace('right') ? leftAdjustment : bottomAdjustment,
+                },
+            };
+
+            return containerPos[position];
+        },
+        [changedPosition, hasSpace, tooltipClient]
+    );
 
+    const getWidth = () => getWidthTooltip(rest.width);
     const width = getWidth();
 
+    const getContainerAlign = () => {
+        if (!tooltipClient) {
+            return;
+        }
+
+        const containerAlign = tooltipClient.width >= minWidth ? 'left' : 'center';
+
+        return containerAlign;
+    };
+
+    const textAlign = getContainerAlign();
```

Comment:
I think you don't need this. If you need to center the content when the tooltip width is `minWidth` (40), you can do that just with css.
```ts
display: 'flex,
flexDirection: 'column',
alignItems: 'center',
```
something like this should do the trick, give it a try